src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -529,23 +522,33 @@ PosixEventEngine::PosixEventEngine(std::shared_ptr<PosixEventPoller> poller)
     : connection_shards_(std::max(2 * gpr_cpu_num_cores(), 1u)),
       executor_(MakeThreadPool(grpc_core::Clamp(gpr_cpu_num_cores(), 4u, 16u))),
 #if GRPC_PLATFORM_SUPPORTS_POSIX_POLLING
-      poller_manager_(poller),
+      scheduler_adapter_(nullptr),
+      poller_(std::move(poller)),
```

Comment:
This relies on `poller` not being nullptr, but that is never asserted. Things will crash if poller == nullptr. You can either assert that here, if you're confident it can never happen, or add runtime checks for nulls.